### PR TITLE
feat(frontend): do not include order keys in materialize pk

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

The distribution should now be correct, though the select statement doesn't produce result. Will fix bugs later.

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests

## Refer to a related PR or issue link (optional)

close https://github.com/singularity-data/risingwave/issues/2099

### CODE REVIEW

**Review by zbzbw — APPROVED** [review]
LGTM

File: src/common/src/catalog/physical_table.rs [hunk 1]
```diff
@@ -18,12 +18,16 @@ use super::{ColumnDesc, OrderedColumnDesc, TableId};
...
 #[derive(Debug, Clone, Default)]
 pub struct TableDesc {
-    /// id of the table, to find in Storage()
+    /// Id of the table, to find in storage.
     pub table_id: TableId,
-    /// the primary key columns' descriptor
-    pub pk: Vec<OrderedColumnDesc>,
-    /// all columns in the table, noticed it is NOT sorted by columnId in the vec
+    /// The keys used to sort in storage.
+    pub order_desc: Vec<OrderedColumnDesc>,
```
Comment (fuyufjh) [inline src/common/src/catalog/physical_table.rs:24]: `order_desc` also has a full description of Column, which is duplicated with `columns`. Perhaps `FieldOrder` is better here?

```
pub struct FieldOrder {
    pub index: usize,
    pub direct: Direction,
}
```

**Review by fuyufjh — APPROVED** [review]
LGTM except the comments

File: src/common/src/catalog/physical_table.rs (hunk 1 above)
Comment (skyzh) [reply src/common/src/catalog/physical_table.rs:24]: Agree. Maybe we can do the refactor and de-dup along with https://github.com/singularity-data/risingwave/issues/2145. Currently I saw `columns` the only source of truth, and `order_desc` is never used to get column info.

**Pull request merged**